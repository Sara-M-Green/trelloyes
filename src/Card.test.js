import React from 'react';
import ReactDOM from 'react-dom';
import CardClass from './Card';
import renderer from 'react-test-renderer';


//test suite
describe('CardClass Component', () => {
    //smoke test
    it('renders without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<CardClass />, div)
        ReactDOM.unmountComponentAtNode(div);
    });

    //snapshot test
    it('renders the UI as expected' , () => {
        const tree = renderer
            .create(<CardClass key="a" card="First card" content="lorem ipsum" />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    });

    //second snapshot test
    it('renders the UI as expected' , () => {
        const tree = renderer
            .create(<CardClass key="g" card="Seventh card" content="lorem ipsum" />)
            .toJSON();
        expect(tree).toMatchSnapshot()
    });
})