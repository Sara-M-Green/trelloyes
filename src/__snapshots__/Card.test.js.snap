// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardClass Component renders the UI as expected 1`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3
    className="Card-title"
  />
  <p
    className="Card-content"
  >
    lorem ipsum
  </p>
</div>
`;

exports[`CardClass Component renders the UI as expected 2`] = `
<div
  className="Card"
>
  <button
    type="button"
  >
    delete
  </button>
  <h3
    className="Card-title"
  />
  <p
    className="Card-content"
  >
    lorem ipsum
  </p>
</div>
`;
